test/e2eprovider/server/server.go

```diff
@@ -0,0 +1,144 @@
+package e2eprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+
+	util "sigs.k8s.io/secrets-store-csi-driver/pkg/csi-common"
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+	types "sigs.k8s.io/secrets-store-csi-driver/test/e2eprovider/types"
+	vault "sigs.k8s.io/secrets-store-csi-driver/test/e2eprovider/vault"
+
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+// Server is a mock csi-provider server
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	SocketPath string
+	network    string
+	Vault      vault.Vault
+}
+
+// NewE2eProviderServer returns a mock csi-provider grpc server
+func NewE2eProviderServer(endpoint string, vault vault.Vault) (*Server, error) {
+	network, address, err := util.ParseEndpoint(endpoint)
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	server := grpc.NewServer()
+	s := &Server{
+		grpcServer: server,
+		SocketPath: address,
+		network:    network,
+		Vault:      vault,
+	}
+
+	v1alpha1.RegisterCSIDriverProviderServer(server, s)
+
+	return s, nil
+}
+
+// Start starts the mock csi-provider server
+func (m *Server) Start() error {
+	var err error
+
+	m.listener, err = net.Listen(m.network, m.SocketPath)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Listening for connections on address: %v", m.listener.Addr())
```

Comment:
Let's use structured logging across all the files to be consistent

Reply:
done